data/src/main/java/cmm/apps/esmorga/data/event/EventRepositoryImpl.kt

```diff
@@ -26,6 +26,11 @@ class EventRepositoryImpl(private val localUserDs: UserDatasource, private val l
         return localEventDs.getEventById(eventId).toEvent()
     }
 
+    override suspend fun joinEvent(eventId: String) {
+        remoteEventDs.joinEvent(eventId)
+        val localEvents = localEventDs.getEvents().map { it.copy(dataUserJoined = it.dataId == eventId || it.dataUserJoined) }
```

Comment:
I think it would be more clear to just change the `dataUserJoined` value on the specific event and not in all of them.